fission-web-server/library/Fission/Web/Server.hs

```diff
@@ -59,8 +58,21 @@ import           Fission.Web.Server.Relay.Store.Class
 import           Fission.Internal.Orphanage.OctetStream      ()
 import           Fission.Internal.Orphanage.PlainText        ()
 
+import qualified Paths_fission_web_server                    as Fission
+
 -- | Top level web API type. Handled by 'server'.
-type API    = Swagger.API :<|> Fission.API :<|> LinkWS
+type API
+  -- V2
+  =    Fission.V2
+  -- Docs
+  :<|> Swagger.Latest
+  :<|> Swagger.V2
+  -- API with omitted docs
+  :<|> Fission.V_
```

Comment:
Why isn't this using `Fission.API` anymore, which includes `Fission.V2` and `Fission.V_`? I assume the ordering makes a difference?

Reply:
The ordering does make a difference, yeah. We want the docs before the unversioned section, because it matches incoming routing in this sequence. For example, if you put the `/` matcher first, nothing else will ever get matched. (This isn't just a servant thing; it's pretty standard)

We have the `servant-swagger` in `fission-web-server` because it imports `servant-server`, which has extra runtime dependencies. EDIT: actually I'm not sure that's true. We don't want to pull in things like Postgres for the CLI, but that's not a dependency for servant-server. I'll take a look at that!

On this topic, I actually have a WIP branch from this weekend that uses [`Servant.API.Generic`](https://www.stackage.org/haddock/lts-18.8/servant-0.18.3/Servant-API-Generic.html), which we had been putting off due. to performance. It's in the main package now, and the compile-time overhead doesn't seem too bad (subjectively, I'm going to run the actual numbers). It cleans up a lot of the router code, but adds a couple potential gotchas (e.g. it fails if you don't use `deriving Generic`). I'll put that up before I merge this.

Reply:
WIP but you can see what this (roughly) looks like here https://github.com/fission-suite/fission/pull/547